internal/controller/ols_app_redis_assets.go

```diff
@@ -70,28 +126,71 @@ func (r *OLSConfigReconciler) generateRedisDeployment(cr *olsv1alpha1.OLSConfig)
 	return &deployment, nil
 }
 
-func (r *OLSConfigReconciler) generateRedisService(cr *olsv1alpha1.OLSConfig) (*corev1.Service, error) {
-	DeploymentSelectorLabels := map[string]string{
-		"app.kubernetes.io/component":  "redis-server",
-		"app.kubernetes.io/managed-by": "lightspeed-operator",
-		"app.kubernetes.io/name":       "lightspeed-service-redis",
-		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+// updateRedisDeployment updates the deployment based on CustomResource configuration.
+func (r *OLSConfigReconciler) updateRedisDeployment(ctx context.Context, existingDeployment, desiredDeployment *appsv1.Deployment) error {
+	changed := false
+
+	// Validate deployment annotations.
+	if existingDeployment.Annotations == nil ||
+		existingDeployment.Annotations[RedisConfigHashKey] != r.stateCache[RedisConfigHashStateCacheKey] {
+		updateDeploymentAnnotations(existingDeployment, map[string]string{
+			RedisConfigHashKey: r.stateCache[RedisConfigHashStateCacheKey],
+		})
+		// update the deployment template annotation triggers the rolling update
+		updateDeploymentTemplateAnnotations(existingDeployment, map[string]string{
+			RedisConfigHashKey: r.stateCache[RedisConfigHashStateCacheKey],
+		})
+
+		changed = true
 	}
+
+	// Update command to a desired value for a specified container.
+	if commandChanged, err := setCommand(existingDeployment, desiredDeployment.Spec.Template.Spec.Containers[0].Command, RedisDeploymentName); err != nil {
+		return err
+	} else if commandChanged {
+		changed = true
+	}
+
+	if changed {
+		r.logger.Info("updating OLS redis deployment", "name", existingDeployment.Name)
+		if err := r.Update(ctx, existingDeployment); err != nil {
+			return err
+		}
+	} else {
+		r.logger.Info("OLS redis deployment reconciliation skipped", "deployment", existingDeployment.Name, "olsconfig hash", existingDeployment.Annotations[RedisConfigHashKey])
+	}
+
+	return nil
+}
+
+func (r *OLSConfigReconciler) generateRedisService(cr *olsv1alpha1.OLSConfig) (*corev1.Service, error) {
+	internalTrafficPolicy := corev1.ServiceInternalTrafficPolicyCluster
+	ipFamilies := []corev1.IPFamily{corev1.IPv4Protocol}
+	ipFamilyPolicy := corev1.IPFamilyPolicySingleStack
 	service := corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      OLSAppRedisServiceName,
+			Name:      RedisServiceName,
 			Namespace: cr.Namespace,
-			Labels:    DeploymentSelectorLabels,
+			Labels:    generateRedisSelectorLabels(),
+			Annotations: map[string]string{
+				"service.beta.openshift.io/serving-cert-secret-name": RedisCertsSecretName,
+			},
 		},
 		Spec: corev1.ServiceSpec{
+			InternalTrafficPolicy: &internalTrafficPolicy,
```

Comment:
Why are we adding this? these are the default values

Reply:
Removed them.